### PR TITLE
Specify email magic link

### PR DESCRIPTION
**Pull request by IniZio**
ref: #2748

### CODE REVIEW

File: docs/specs/user-model.md [hunk 1]
```diff
@@ -374,6 +375,27 @@ which OOB-OTP authenticator to use when performing authentication. However, a
...
+
+Magic link is a sub-category under OOB-OTP authenticator. User receives a login link instead of OTP code, which authenticates user on visit.
+
+Currently magic link is only available as secondary authenticator, and can be setup without verification. It will become verified once user successfully authenticates with magic link.
+
+```yaml
+authentication:
+  oob_otp:
+    email:
+      seconday_email_otp_mode: "magic_link"
```
Comment (IniZio) [inline docs/specs/user-model.md:388]: Not sure if we will want to allow a "email_otp_mode" option as well?

File: docs/specs/event.md [hunk 2]
```diff
@@ -27,10 +27,12 @@
       - [authentication.primary.password.failed](#authenticationprimarypasswordfailed)
       - [authentication.primary.oob_otp_email.failed](#authenticationprimaryoob_otp_emailfailed)
       - [authentication.primary.oob_otp_sms.failed](#authenticationprimaryoob_otp_smsfailed)
+      - [authentication.primary.oob_otp_email_magic_link.failed](#authenticationprimaryoob_otp_email_magic_linkfailed)
```
Comment (louischan-oursky) [inline docs/specs/event.md:30]: Remove this event.

File: docs/specs/event.md [hunk 3]
```diff
@@ -27,10 +27,12 @@
...
       - [authentication.secondary.password.failed](#authenticationsecondarypasswordfailed)
       - [authentication.secondary.totp.failed](#authenticationsecondarytotpfailed)
       - [authentication.secondary.oob_otp_email.failed](#authenticationsecondaryoob_otp_emailfailed)
       - [authentication.secondary.oob_otp_sms.failed](#authenticationsecondaryoob_otp_smsfailed)
+      - [authentication.secondary.oob_otp_email_magic_link.failed](#authenticationsecondaryoob_otp_email_magic_linkfailed)
```
Comment (louischan-oursky) [inline docs/specs/event.md:35]: Remove this event.

File: docs/specs/templates.md [hunk 4]
```diff
@@ -126,3 +127,22 @@ One-time-password message. Used for authentication and user verification.
...
+- Context:
+    - `AppName`: the display name of the app.
+    - `Email`: The recipient email of the message.
+    - `Phone`: The recipient phone number of the message.
+    - `LoginID`: The login ID of the identity
+        - `LoginID.Key`: Login ID key
+        - `LoginID.Value`: Login ID
+    - `Link`: The magic link.
+    - `Host`: Host of authgear.
+    - `Origin`: The origin page of the magic link, can be `signup`/`login`/`settings`.
```
Comment (louischan-oursky) [inline docs/specs/templates.md:147]: Refer to pkg/lib/authn/otp/template.go MessageTemplateContext to see what variables are actually available. Remove anything that is not implemented.

File: docs/specs/user-model.md (hunk 1 above)
Comment (louischan-oursky) [reply docs/specs/user-model.md:388]: No need for now.

File: docs/specs/user-model.md [hunk 5]
```diff
@@ -374,6 +375,27 @@ which OOB-OTP authenticator to use when performing authentication. However, a
...
+      seconday_allow_unverified: true
+
+verification:
+  magic_link_expiry_seconds: 3600
```
Comment (louischan-oursky) [inline docs/specs/user-model.md:392]: No need to add this config. We are going to use the expiry of OTP.

**Comment by IniZio** [comment]
@louischan-oursky Updated 🙏

**Comment by IniZio** [comment]
Updated to specify when can skip verification (only signup) and when can magic link is available (only login) also

**Comment by IniZio** [comment]
As per discussion, OTP code will be sent together with magic link. Will update the spec accoridingly.

**Comment by IniZio** [comment]
Updated to reflect decision of having login link available as primary authenticator

**Pull request merged**